.markdownlint.yml

```diff
@@ -0,0 +1,16 @@
+default: true
```

Comment:
Is there any way this file can be moved to `.github/workflow-config/.markdownlint.yml` and have an option added to the runner to load the config file from there in the same way we have done for ansible-lint?

Reply:
I can, but the advantage of having the config file in the root folder is that users using linters in an IDE (such as VSCode) will use the config when editing files. If the config file is in a different directory it isn't picked up by the IDE linters by default. I was going to propose that the other linter configs be moved to the root folder for this reason. 

IMHO it is a better workflow for the user to be able to see linter errors in the IDE before pushing a commit as opposed to getting the errors after pushing. Another argument for this setup is that GitHub Actions is limited to displaying 10 annotations for each step and 50 per 50 (https://github.community/t/annotation-limitation/17998/2), so if there are more than 10 linter errors the user won't be able to see them unless they look at the details of the action run.

Reply:
Bringing it into pre-commit I don't think is a problem. Pre-commit and the tools being run can use config files in the root folder. In fact the "Hack" that is in place to copy the pre-commit config from the workflow-config folder to the root folder could be eliminated as well if the config files were move to the root folder. The config files should be added to the galaxy.xml.j2 so they are not included in the collection build.

Reply:
I'm back and forth on this, We definitely want to add it to the Galaxy build ignore files, but with IDE things like this reading automatically, and some of the linters wanting it in base, it might be time to move it to the TLD and leave it there, we initially moved it to the sub folders just to keep things 'clean', and I think the disadvantages are starting to outweigh the advantage of that.

Reply:
TLD, my vote is to move it to base, as its giving more advantages then it did 2 years ago.

Reply:
:+1: Sounds good. I'll move them to TLD.

Reply:
I changed the markdownlint to use the pre-commit action and moved all the linter configs to TLD. I was also able to remove the separate yamllint and just let ansible-lint perform the yamllint as it now picks up the .yamllint.yml which extends its configuration. Some lint errors have also been fixed.

Reply:
@Tompage1994 Handing this over to you for final approval, I think its at a good place, but with the TLD change, and everything going on, want you to at least have a chance to review before we merge.